src/connector/src/schema/schema_registry/client.rs

```diff
@@ -61,29 +61,44 @@ pub struct Client {
     password: Option<String>,
 }
 
+#[derive(Debug, thiserror::Error)]
+#[error("all request confluent registry all timeout, {context}\n{}", errs.iter().map(|e| format!("\t{}", e.as_report())).join("\n"))]
+pub struct ConcurrentRequestError {
+    errs: Vec<itertools::Either<RequestError, tokio::task::JoinError>>,
+    context: String,
+}
+
+impl From<ConcurrentRequestError> for risingwave_common::error::RwError {
+    fn from(value: ConcurrentRequestError) -> Self {
+        anyhow::anyhow!(value).into()
+    }
+}
+
+type SrResult<T> = Result<T, ConcurrentRequestError>;
+
 impl Client {
-    pub(crate) fn new(url: Vec<Url>, client_config: &SchemaRegistryAuth) -> Result<Self> {
+    pub(crate) fn new(
+        url: Vec<Url>,
+        client_config: &SchemaRegistryAuth,
+    ) -> Result<Self, InvalidOptionError> {
         let valid_urls = url
             .iter()
             .map(|url| (url.cannot_be_a_base(), url))
             .filter(|(x, _)| !*x)
             .map(|(_, url)| url.clone())
             .collect_vec();
         if valid_urls.is_empty() {
-            return Err(RwError::from(ProtocolError(format!(
-                "no valid url provided, got {:?}",
-                url
-            ))));
+            return Err(invalid_option_error!("non-base: {}", url.iter().join(" ")));
         } else {
             tracing::debug!(
                 "schema registry client will use url {:?} to connect",
                 valid_urls
             );
         }
 
-        let inner = reqwest::Client::builder().build().map_err(|e| {
-            RwError::from(ProtocolError(format!("build reqwest client failed {}", e)))
-        })?;
+        let inner = reqwest::Client::builder()
+            .build()
+            .map_err(|e| invalid_option_error!("build reqwest client failed: {}", e.as_report()))?;
```

Comment:
Shall `invalid_option_error!` macro accept `reqwest::Error` (or `Into<BoxedError>`) as `#[source]` field?

Reply:
That would be better. The source error could be displayed in a multi-line manner in pretty mode.

Reply:
Updated to `unwrap` for this specific case. The call is not affected by any input.